### PR TITLE
Avoid tenant discovery recursion

### PR DESCRIPTION
**Pull request by mansam**
https://bugzilla.redhat.com/show_bug.cgi?id=1561637

### CODE REVIEW

File: lib/manageiq/providers/openstack/legacy/openstack_handle/handle.rb [hunk 1]
```diff
@@ -380,7 +382,7 @@ def accessible_tenant_names
     def default_tenant_name
       return @default_tenant_name ||= "admin" if tenant_accessible?("admin")
       tenant_names.each do |name|
-        next if name == "services"
+        next if name.blank? || name == "services"
```
Comment (aufi) [inline lib/manageiq/providers/openstack/legacy/openstack_handle/handle.rb]: Is there an example for case when name.blannk? == true? I've left it in example code from experimenting with tenant with empty name, but I haven't found such case in OSP.

Comment (mansam) [reply lib/manageiq/providers/openstack/legacy/openstack_handle/handle.rb]: I didn't think so, but I thought perhaps you had a good reason for it. I'll pull the change if you agree it's unlikely to come up.

Comment (aufi) [reply lib/manageiq/providers/openstack/legacy/openstack_handle/handle.rb]: Yep, I'd suggest drop it in that case.

**Comment by aufi** [comment]
Looks good to me :+1:

**Comment by simaishi** [comment]
@mansam Most of the VCR cassettes are conflicting during backport to gaprindashvili. Should I just take the VCRs from this PR ignoring the conflicts, or do you need to re-generate for the G branch?

**Comment by mansam** [comment]
@simaishi Unfortunately I have to regenerate them for the G branch. I'll create a new PR and then ping you.

**Comment by simaishi** [comment]
Backported to Gaprindashvili via https://github.com/ManageIQ/manageiq-providers-openstack/pull/268

**Pull request merged**